src/python/dxpy/cli/dataset_utilities.py

```diff
@@ -0,0 +1,559 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 DNAnexus, Inc.
+#
+# This file is part of dx-toolkit (DNAnexus platform client libraries).
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#   use this file except in compliance with the License. You may obtain a copy
+#   of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+from __future__ import print_function, unicode_literals, division, absolute_import
+
+import sys
+import collections
+import json
+import os
+import re
+import csv
+import dxpy
+import pandas as pd
+from ..utils.printing import (fill)
+from ..bindings import DXRecord
+from ..bindings.dxdataobject_functions import is_dxlink
+from ..bindings.dxfile import DXFile
+from ..utils.resolver import resolve_existing_path, ResolutionError
+from ..utils.file_handle import as_handle
+from ..exceptions import err_exit
+
+database_unique_name_regex = re.compile('^database_\w{24}__\w+$')
+database_id_regex = re.compile('^database-\\w{24}$')
+
+def extract_dataset(args):
+    """
+       (Preview feature) Retrieves the data, or generates SQL to retrieve the data, from a dataset or cohort for a set of entity.fields. 
+    """
+    if not args.dump_dataset_dictionary and args.fields is None:
+        raise err_exit(fill('Must provide at least one of the following options: --fields or --dump-dataset-dictionary'))
+
+    if len(args.delim) == 1:
+        delimiter = str(args.delim)
+        if delimiter == ",":
+            out_extension = ".csv"
+        elif delimiter == "\t":
+            out_extension = ".tsv"
+        else:
+            out_extension = ".txt"
+    else:
+        raise err_exit(fill('Invalid delimiter specified'))
+    
+    project, path, entity_result = resolve_existing_path(args.path)
+
+    try:
+        resp = dxpy.DXHTTPRequest('/' + entity_result['id'] + '/visualize',
+                                        {"project": project, "cohortBrowser": False} )
+    except Exception as details:
+        if details.name == 'PermissionDenied':
+            print("Insufficient permissions")
+            sys.exit(1)
+        elif details.name in ["InvalidInput", "InvalidState"]:
+            print('%r : Invalid cohort or dataset' % entity_result['id'])
+            sys.exit(1)
+        else:
+            raise ResolutionError(str(details))
```

Comment:
Minor: I would suggest to use multiple `except` statements for clarity and also to avoid `name` checking as string (like https://stackoverflow.com/questions/6095717/python-one-try-multiple-except)

Reply:
Updated! Thank you Jan.